services/brig/src/Brig/API/Public.hs

```diff
@@ -328,13 +328,22 @@ type GetUserClientQualified =
     :> Get '[Servant.JSON] Public.PubClient
 
 type ListClientsBulk =
-  Summary "List all clients for a set of user ids"
+  Summary "List all clients for a set of user ids (deprecated, use /user/list-clients/v2)"
     :> ZAuthServant
     :> "users"
     :> "list-clients"
     :> Servant.ReqBody '[Servant.JSON] (Range 1 MaxUsersForListClientsBulk [Qualified UserId])
     :> Post '[Servant.JSON] (Public.QualifiedUserMap (Set Public.PubClient))
 
+type ListClientsBulkV2 =
+  Summary "List all clients for a set of user ids"
+    :> ZAuthServant
+    :> "users"
+    :> "list-clients"
+    :> "v2"
+    :> Servant.ReqBody '[Servant.JSON] (Public.LimitedQualifiedUserIdList MaxUsersForListClientsBulk)
+    :> Post '[Servant.JSON] (Public.QualifiedUserMap (Set Public.PubClient))
```

Comment:
While we're at it: is giving back a map a great idea? Wouldn't we have more flexibility for the future if we give back an object containing the map here: `{"qualifiedUserMap" : { "domain" : [ ... ] }}`? 
(feel free to ignore me here and merge the PR as is. I just thought I'd bring that idea up)

Reply:
Hmm, I am going to implement the `Named` thing I did lazily, but in a slightly better way and in wire-api to achieve this.

Reply:
Done, LMK what you think.